tooling/nargo_cli/build.rs

```diff
@@ -744,3 +745,71 @@ fn plonky2_verify_success_{test_name}() {{
         .expect("Could not write templated test file.");
     }
 }
+
+/// Tests using the experimental PLONKY2 backend as a proving engine that are expected to fail verification.
+fn generate_plonky2_verify_failure_tests(test_file: &mut File, test_data_dir: &Path) {
+    let test_sub_dir = "plonky2_verify_failure";
+    let test_data_dir = test_data_dir.join(test_sub_dir);
+
+    let test_case_dirs =
+        fs::read_dir(test_data_dir).unwrap().flatten().filter(|c| c.path().is_dir());
+
+    let expected_messages = HashMap::from([("simple_add", vec!["Cannot satisfy constraint"])]);
```

Comment:
You could use this to check the exact error message.

It would be ("zk_dungeon_verify_fail_1", vec!["VerificationFailed { message: \"Public inputs don't match proof\" }."])

Alternatively, you could just remove this and related mechanism for checking messages match.

Reply:
Done. Unfortunately untested, because the tests fail, due to upstream breakage of the verify command.